src/main/java/org/broadinstitute/hellbender/engine/ReadWalker.java

```diff
@@ -57,7 +57,7 @@ protected final void onStartup() {
      */
     void setReadTraversalBounds() {
         if ( hasUserSuppliedIntervals() ) {
-            reads.setTraversalBounds(intervalArgumentCollection.getTraversalParameters(getHeaderForReads().getSequenceDictionary()));
+            reads.setTraversalBounds(userIntervals);
```

Comment:
@tedsharpe Not sure if this was a temporary workaround for some issue you encountered, but the previous implementation passing in `intervalArgumentCollection.getTraversalParameters()` is actually necessary to support `-L unmapped`. Hopefully this is covered by existing tests somewhere...

Reply:
Oh, bummer.  I'll have to rework this, then.  I'm a little vague on the details right now, but it had something to do with an additional tier of padding that I'm doing to make sure that I pull in the distant mates (via the tool whose design you suggested).  I had to expand the intervals over which I do read retrieval, while still having access to the underlying singly-padded intervals over which I do assembly.
I asked Louis about why we pre-calculated userIntervals if we were just going to recalculate them here.  He thought it seemed an obvious bug at the time.  It's so hard to change engine code.  :-(

Reply:
Don't despair @tedsharpe, I think we can probably fix this pretty easily! You just need to construct a new `TraversalParameters` here, and make sure that the user's `traverseUnmapped` setting gets propagated. Something like this could do for now:

```
    void setReadTraversalBounds() {
        if ( hasUserSuppliedIntervals() ) {
            reads.setTraversalBounds(new TraversalParameters(userIntervals, intervalArgumentCollection.getTraversalParameters(getHeaderForReads().getSequenceDictionary()).traverseUnmappedReads());
        }
    }
```